simulator/simulator.go

```diff
@@ -35,11 +37,23 @@ func startSimulator() error {
 		return xerrors.Errorf("get config: %w", err)
 	}
 
-	restclientCfg, apiShutdown, err := k8sapiserver.StartAPIServer(cfg.KubeAPIServerURL, cfg.EtcdURL, cfg.CorsAllowedOriginList)
-	if err != nil {
-		return xerrors.Errorf("start API server: %w", err)
+	kubeAPIServerURL := cfg.KubeAPIServerURL
```

Comment:
KubeAPIServerURL has the default value. 
So, you gotta change something if you'd like to reuse this.

https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/0be3194647ec86ec8e0b3befb94c015ddf366c0c/simulator/config/config.go#L139-L156

Or, probably just adding `EnableKubeAPIServer` parameter to the config is the simplest way?

Reply:
> Or, probably just adding EnableKubeAPIServer parameter to the config is the simplest way?

It looks good to me. How about like this ? We remove KubeAPIServer URL default value or remove KubeAPIServerURL field , add a new receiver method to return the string of KubeAPIServerURL. I think it's simple way. What do you think ?

Reply:
I don't quite get what you mean. You mean creating another env named `KubeAPIServerURL` and remove default values from `KUBE_API_PORT` and `KUBE_API_HOST`? (If wrong, please elaborate.

Reply:
Sorry, I don't know what I was getting at, too. Please forget it once.
I think it's good way to add new env named ExternalKubeAPIServerURL. If is's set, a simulator uses external kube-api-server. If not, a simulator uses internal kube-api-server which creates itself. What do you think ?

Reply:
sg